packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -54,20 +54,18 @@ function logError(boundary: Fiber, errorInfo: CapturedValue<mixed>) {
   const capturedError: CapturedError = {
     componentName: source !== null ? getComponentName(source) : null,
     error: errorInfo.value,
-    errorBoundary: boundary,
+    errorBoundary: null,
     componentStack: stack !== null ? stack : '',
     errorBoundaryName: null,
     errorBoundaryFound: false,
     willRetry: false,
   };
 
   if (boundary !== null) {
+    capturedError.errorBoundary = boundary.stateNode;
     capturedError.errorBoundaryName = getComponentName(boundary);
     capturedError.errorBoundaryFound = capturedError.willRetry =
```

Comment:
This line still confuses me a bit. Can it be something other than `ClassComponent`? Should the `errorBoundary` assignment also be conditional on that?

Reply:
`componentDidCatch` only works for class component API.

Reply:
Right, which is why the `tag` check on the next line confused me.

Reply:
https://github.com/gaearon/react/blob/509b257a30839b6b47f53ae522679756161cdf84/packages/react-reconciler/src/ReactFiberCommitWork.js#L291